Replace seam taglib

huangp opened the pull request:
https://zanata.atlassian.net/browse/ZNTA-592

seanf commented:
@huangp If you rebase this branch on top of master after #908 goes in, GitHub should hide all the previously reviewed security commits.  Although if you squash-merge #908, the rebase might be a bit more difficult, and it might be easier to cherry pick a range of commits.

huangp commented:
@seanf sorry I rebased this branch on top of master with squash merged #908. It lost all your previous comments. But It's just the first two commits you have reviewed.

huangp commented:
build passed http://zanata-build.lab.eng.bne.redhat.com/job/zanata-server-github-pull-requests/4250/

seanf commented:
:+1:

definite commented:
@zanata-jenkins retest this please

djansen-redhat commented on functional-test/src/main/java/org/zanata/page/account/InactiveAccountPage.java:
```diff
@@ -43,12 +43,13 @@ public HomePage clickResendActivationEmail() {
     public InactiveAccountPage enterNewEmail(String email) {
-        enterText(readyElement(By.id("inactiveAccountForm:emailField:email")), email);
+        enterText(readyElement(By.id("inactiveAccountForm:email:input:email"))
+                , email);
```
CommaOnNextLineException

djansen-redhat commented:
- [ ] The "Clear" function on the new Glossary upload doesn't actually clear. I don't know if this matters.

djansen-redhat commented:
Loading the Server Config page works, but shows a lot of this in the log:
2015-09-16T23:55:48,724Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID registerUrl in view.
2015-09-16T23:55:48,731Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID adminEml in view.
2015-09-16T23:55:48,735Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID fromEml in view.
2015-09-16T23:55:48,739Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID termsOfUseUrlEml in view.
2015-09-16T23:55:48,744Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID helpInput in view.
2015-09-16T23:55:48,768Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID logDestEml in view.
2015-09-16T23:55:48,776Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID maxConcurrentPerApiKeyEml in view.
2015-09-16T23:55:48,779Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID maxActiveRequestsPerApiKeyEml in view.
2015-09-16T23:55:48,782Z WARNING [javax.enterprise.resource.webcontainer.jsf.renderkit] (default task-23) Unable to find component with ID maxFilesPerUploadFieldInput in view.

seanf commented:
Commit b70d720 is cherry-picked in https://github.com/zanata/zanata-server/pull/973

seanf commented:
- [ ] I just searched for `http://jboss.org/schema/seam/taglib` and found about 80 hits.  And there are a couple of references to `s:link` in the xhtml files too.  Could you clean these up please?

seanf commented:
- [ ] I think someone with more JSF experience ought to review this too.

huangp commented:
```http://jboss.org/schema/seam/taglib``` is in namespace and we still have them there because some file still uses seam tag. e.g. s:token and s:conversationId. s:link is only in 
```
<a href="#" jsfc="s:link"
```
@carlosmunoz do you know what this means?

seanf commented:
I've removed the taglib references except where s: is still in use.  See
commit 07633a3.

On 17 September 2015 at 16:51, Patrick Huang <notifications@github.com>
wrote:

> http://jboss.org/schema/seam/taglib is in namespace and we still have
> them there because some file still uses seam tag. e.g. s:token and
> s:conversationId. s:link is only in
>
> <a href="#" jsfc="s:link"
>
> @carlosmunoz <https://github.com/carlosmunoz> do you know what this means?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928#issuecomment-140986064>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

aeng commented on functional-test/pom.xml:
```diff
@@ -516,6 +516,7 @@
               <classifier>mysql.classifier</classifier>
               <port>${mysql.port}</port>
               <socket>${mysql.socket}</socket>
+              <clearexistingdata>false</clearexistingdata>
```
CamelCase?

huangp replied:
it is all lower case according to the plugin documentation

On Wed, Sep 23, 2015 at 1:50 PM, Alex Eng <notifications@github.com> wrote:

> In functional-test/pom.xml
> <https://github.com/zanata/zanata-server/pull/928#discussion_r40169186>:
>
> > @@ -516,6 +516,7 @@
> >                <classifier>mysql.classifier</classifier>
> >                <port>${mysql.port}</port>
> >                <socket>${mysql.socket}</socket>
> > +              <clearexistingdata>false</clearexistingdata>
>
> CamelCase?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928/files#r40169186>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

aeng commented on zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java:
```diff
@@ -734,18 +740,22 @@ public void uploadSourceFile() {
+                        +
+                        sourceFileUpload.getFileName();
+                // TODO this message is not displayed
                 conversationScopeMessages.setMessage(
```
Should `conversationScopeMessages` gets replace?

huangp replied:
It should. But not in this PR.

On Wed, Sep 23, 2015 at 1:53 PM, Alex Eng <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/action/VersionHomeAction.java
> <https://github.com/zanata/zanata-server/pull/928#discussion_r40169265>:
>
> >              } else {
> > +                String summary = "Unrecognized file extension for "
> > +                        +
> > +                        sourceFileUpload.getFileName();
> > +                // TODO this message is not displayed
> >                  conversationScopeMessages.setMessage(
>
> Should conversationScopeMessages gets replace?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928/files#r40169265>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/dashboard/settings.xhtml:
```diff
@@ -211,11 +209,11 @@
           <div class="form__item">
             <h:outputLabel for="nameField">#{msgs['jsf.Name']}</h:outputLabel>
-            <s:decorate id="nameField" template=" ../field.xhtml" enclose="true">
+            <zanata:decorate id="nameField" componentId="name">
```
`componentId='accountName'` ?

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/admin/contact_admin_modal.xhtml:
```diff
@@ -57,12 +57,12 @@
 
       <div class="modal__content l--pad-all-1">
-        <s:decorate id="messageField" template="/WEB-INF/layout/field.xhtml">
+        <zanata:decorate id="messageField" componentId="message">
```
`componentId='contact-admin-message'`?

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/admin/contact_admin_modal_anonymous.xhtml:
```diff
@@ -58,10 +58,10 @@
         <span>#{msgs['jsf.AdditionalInfo']}</span>
         <span class="txt--meta bx--block">#{msgs['jsf.passwordReset.google.AdditionalInfoMessage']}</span>
-        <s:decorate id="messageField" template="/WEB-INF/layout/field.xhtml">
+        <zanata:decorate id="messageField">
```
is componentId required here?

huangp replied:
Only if you want to a4j on blur event. It needs to know which id to
re-render. This one doesn't have a4j in there so not needed.

On Wed, Sep 23, 2015 at 2:08 PM, Alex Eng <notifications@github.com> wrote:

> In
> zanata-war/src/main/webapp/WEB-INF/layout/admin/contact_admin_modal_anonymous.xhtml
> <https://github.com/zanata/zanata-server/pull/928#discussion_r40169701>:
>
> > @@ -58,10 +58,10 @@
> >        <div class="modal__content l--pad-all-1">
> >          <span>#{msgs['jsf.AdditionalInfo']}</span>
> >          <span class="txt--meta bx--block">#{msgs['jsf.passwordReset.google.AdditionalInfoMessage']}</span>
> > -        <s:decorate id="messageField" template="/WEB-INF/layout/field.xhtml">
> > +        <zanata:decorate id="messageField">
>
> is componentId required here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928/files#r40169701>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/language/contact_coordinator_modal.xhtml:
```diff
@@ -50,12 +50,12 @@
 
       <div class="modal__content l--pad-all-1">
-        <s:decorate id="messageField" template="/WEB-INF/layout/field.xhtml">
+        <zanata:decorate id="messageField" componentId="message">
```
componentId='contact-coordinator-message'?

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/language/contact_coordinator_modal.xhtml:
```diff
@@ -50,12 +50,12 @@
           <h:inputTextarea id="contact-coordinator-message" value="#{languageContactCoordinatorAction.message}"
             styleClass="l--push-bottom-0">
-            <a4j:ajax event="blur" execute="@this"/>
+            <a4j:ajax event="blur" render="message" execute="@this"/>
```
render='contact-coordinator-message'

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/language/request_join_modal.xhtml:
```diff
@@ -47,35 +47,35 @@
         </div>
 
-        <s:decorate id="messageField" template="/WEB-INF/layout/field.xhtml">
+        <zanata:decorate id="message" componentId="messageField">
```
componentId='request-join-message'?

huangp replied:
componentId need to match the render value inside ```<a4j:ajax event="blur" render="messageField" execute="@this"/>```. I tried to minimize changes so reused the old id here. Same with other places

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/language/settings-tab.xhtml:
```diff
@@ -30,13 +29,13 @@
               title="#{msgs.format('jsf.language.displayName.placeholder',languageAction.locale.retrieveDefaultDisplayName())}"></i>
           </span>
-          <s:decorate id="nameField" template="../field.xhtml" enclose="true">
+          <zanata:decorate id="name" componentId="nameField">
```
componentId='name'?

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/language/settings-tab.xhtml:
```diff
@@ -82,14 +81,14 @@
             <i class="i i--help"></i>
           </a>
-          <s:decorate id="pluralFormsField" template="../field.xhtml" enclose="true">
+          <zanata:decorate id="pluralForms" componentId="pluralFormsField">
```
componentId='pluralForms'

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/project/edit_form.xhtml:
```diff
@@ -24,11 +24,11 @@
       <div class="form__item">
         <h:outputLabel for="name">#{msgs['jsf.ProjectName']}</h:outputLabel>
-        <s:decorate id="nameField" template="../field.xhtml" enclose="true">
+        <zanata:decorate id="name" componentId="nameField">
```
componentId="name"

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/project/edit_form.xhtml:
```diff
@@ -38,7 +38,7 @@
             title="#{msgs['jsf.SlugExample']}"></i></span>
         </h:outputLabel>
-        <s:decorate id="slugField" template="../field.xhtml" enclose="true">
+        <zanata:decorate id="slug" componentId="slugField">
```
componentId="slug"

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/project/edit_form.xhtml:
```diff
@@ -56,12 +56,12 @@
       <span class="txt--meta">#{msgs['jsf.field.optional']}</span>
     </h:outputLabel>
-    <s:decorate id="descriptionField" template="../field.xhtml">
+    <zanata:decorate id="description" componentId="descriptionField">
```
componentId="description"

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/project/settings-tab-general.xhtml:
```diff
@@ -38,13 +39,13 @@
           <span class="txt--meta">#{msgs['jsf.field.optional']}</span>
         </h:outputLabel>
-        <s:decorate id="homePageField" template="../field.xhtml">
+        <zanata:decorate id="homePage" componentId="homePageField">
```
componentId="homePage"

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/project/settings-tab-general.xhtml:
```diff
@@ -57,12 +58,12 @@
           <span class="txt--meta">#{msgs['jsf.field.optional']}</span>
         </h:outputLabel>
-        <s:decorate id="repoField" template="../field.xhtml">
+        <zanata:decorate id="repo" componentId="repoField">
```
componentId="repo"

djansen-redhat commented:
Getting a lot of this on copytrans:
```
2015-09-23T04:14:07,166Z WARN  [org.zanata.service.impl.CopyTransServiceImpl] (pool-5-thread-1) exception during copy trans: java.lang.NullPointerException
	at org.zanata.service.impl.CopyTransWork.shouldFindMatch(CopyTransWork.java:297) [classes:]
	at org.zanata.service.impl.CopyTransWork.work(CopyTransWork.java:163) [classes:]
	at org.zanata.service.impl.CopyTransWork.work(CopyTransWork.java:36) [classes:]
	at org.jboss.seam.util.Work.workInTransaction(Work.java:61) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.zanata.service.impl.CopyTransServiceImpl.copyTransForBatch(CopyTransServiceImpl.java:192) [classes:]
	at org.zanata.service.impl.CopyTransServiceImpl.copyTransForDocumentLocale(CopyTransServiceImpl.java:134) [classes:]
	at org.zanata.service.impl.CopyTransServiceImpl.copyTransForDocument(CopyTransServiceImpl.java:243) [classes:]
	at org.zanata.service.impl.CopyTransServiceImpl.copyTransForIteration(CopyTransServiceImpl.java:294) [classes:]
	at org.zanata.service.impl.CopyTransServiceImpl.startCopyTransForIteration(CopyTransServiceImpl.java:261) [classes:]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_60]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_60]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_60]
	at java.lang.reflect.Method.invoke(Method.java:497) [rt.jar:1.8.0_60]
	at org.jboss.seam.util.Reflections.invoke(Reflections.java:22) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.RootInvocationContext.proceed(RootInvocationContext.java:32) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:56) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.zanata.async.AsyncMethodInterceptor.aroundInvoke(AsyncMethodInterceptor.java:107) [classes:]
	at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.transaction.RollbackInterceptor.aroundInvoke(RollbackInterceptor.java:28) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.core.BijectionInterceptor.aroundInvoke(BijectionInterceptor.java:79) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.core.MethodContextInterceptor.aroundInvoke(MethodContextInterceptor.java:44) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.RootInterceptor.invoke(RootInterceptor.java:107) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.JavaBeanInterceptor.interceptInvocation(JavaBeanInterceptor.java:196) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.jboss.seam.intercept.JavaBeanInterceptor.invoke(JavaBeanInterceptor.java:114) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
	at org.zanata.service.impl.CopyTransServiceImpl_$$_javassist_seam_96.startCopyTransForIteration(CopyTransServiceImpl_$$_javassist_seam_96.java) [classes:]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_60]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_60]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_60]
	at java.lang.reflect.Method.invoke(Method.java:497) [rt.jar:1.8.0_60]
	at org.zanata.async.AsyncMethodInterceptor$1.call(AsyncMethodInterceptor.java:84) [classes:]
	at org.zanata.async.AsyncTaskManager$1.execute(AsyncTaskManager.java:100) [classes:]
	at org.zanata.security.ZanataIdentity.runAs(ZanataIdentity.java:645) [classes:]
	at org.zanata.async.AsyncTaskManager$RunnableOperation.run(AsyncTaskManager.java:164) [classes:]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_60]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_60]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_60]
```

aeng commented on zanata-war/src/main/webapp/WEB-INF/layout/version-group/projects-tab.xhtml:
```diff
@@ -162,9 +161,9 @@
         id="projectsTab-project-label-loader"/>
 
-      <s:span id="project-label" styleClass="i__text--right">
+      <h:panelGroup id="project-label" class="i__text--right">
```
styleClass?

huangp replied:
Good catch. Fixed all of them

On Wed, Sep 23, 2015 at 2:24 PM, Alex Eng <notifications@github.com> wrote:

> In
> zanata-war/src/main/webapp/WEB-INF/layout/version-group/projects-tab.xhtml
> <https://github.com/zanata/zanata-server/pull/928#discussion_r40170157>:
>
> > @@ -162,9 +161,9 @@
> >        <zanata:loader jsHandle="true" layout="inline" type="loader--small" visible="false"
> >          id="projectsTab-project-label-loader"/>
> >
> > -      <s:span id="project-label" styleClass="i__text--right">
> > +      <h:panelGroup id="project-label" class="i__text--right">
>
> styleClass?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928/files#r40170157>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com

seanf commented:
When testing this branch, you mean?  Any reason you CCed me in particular?

On 23 September 2015 at 14:19, Damian Jansen <notifications@github.com>
wrote:

> Getting a lot of this on copytrans:
>
> 2015-09-23T04:14:07,166Z WARN  [org.zanata.service.impl.CopyTransServiceImpl] (pool-5-thread-1) exception during copy trans: java.lang.NullPointerException
>     at org.zanata.service.impl.CopyTransWork.shouldFindMatch(CopyTransWork.java:297) [classes:]
>     at org.zanata.service.impl.CopyTransWork.work(CopyTransWork.java:163) [classes:]
>     at org.zanata.service.impl.CopyTransWork.work(CopyTransWork.java:36) [classes:]
>     at org.jboss.seam.util.Work.workInTransaction(Work.java:61) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.zanata.service.impl.CopyTransServiceImpl.copyTransForBatch(CopyTransServiceImpl.java:192) [classes:]
>     at org.zanata.service.impl.CopyTransServiceImpl.copyTransForDocumentLocale(CopyTransServiceImpl.java:134) [classes:]
>     at org.zanata.service.impl.CopyTransServiceImpl.copyTransForDocument(CopyTransServiceImpl.java:243) [classes:]
>     at org.zanata.service.impl.CopyTransServiceImpl.copyTransForIteration(CopyTransServiceImpl.java:294) [classes:]
>     at org.zanata.service.impl.CopyTransServiceImpl.startCopyTransForIteration(CopyTransServiceImpl.java:261) [classes:]
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_60]
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_60]
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_60]
>     at java.lang.reflect.Method.invoke(Method.java:497) [rt.jar:1.8.0_60]
>     at org.jboss.seam.util.Reflections.invoke(Reflections.java:22) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.RootInvocationContext.proceed(RootInvocationContext.java:32) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:56) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.zanata.async.AsyncMethodInterceptor.aroundInvoke(AsyncMethodInterceptor.java:107) [classes:]
>     at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.transaction.RollbackInterceptor.aroundInvoke(RollbackInterceptor.java:28) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.core.BijectionInterceptor.aroundInvoke(BijectionInterceptor.java:79) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.core.MethodContextInterceptor.aroundInvoke(MethodContextInterceptor.java:44) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.SeamInvocationContext.proceed(SeamInvocationContext.java:68) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.RootInterceptor.invoke(RootInterceptor.java:107) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.JavaBeanInterceptor.interceptInvocation(JavaBeanInterceptor.java:196) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.jboss.seam.intercept.JavaBeanInterceptor.invoke(JavaBeanInterceptor.java:114) [jboss-seam-2.3.1.Final.jar:2.3.1.Final]
>     at org.zanata.service.impl.CopyTransServiceImpl_$$_javassist_seam_96.startCopyTransForIteration(CopyTransServiceImpl_$$_javassist_seam_96.java) [classes:]
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.8.0_60]
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) [rt.jar:1.8.0_60]
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.8.0_60]
>     at java.lang.reflect.Method.invoke(Method.java:497) [rt.jar:1.8.0_60]
>     at org.zanata.async.AsyncMethodInterceptor$1.call(AsyncMethodInterceptor.java:84) [classes:]
>     at org.zanata.async.AsyncTaskManager$1.execute(AsyncTaskManager.java:100) [classes:]
>     at org.zanata.security.ZanataIdentity.runAs(ZanataIdentity.java:645) [classes:]
>     at org.zanata.async.AsyncTaskManager$RunnableOperation.run(AsyncTaskManager.java:164) [classes:]
>     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [rt.jar:1.8.0_60]
>     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [rt.jar:1.8.0_60]
>     at java.lang.Thread.run(Thread.java:745) [rt.jar:1.8.0_60]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/928#issuecomment-142486518>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

aeng commented:
:+1:

djansen-redhat commented:
@seanf that comment was directed at the author

huangp commented:
- fixed clear on glossary upload and other two uploads
- fixed class to styleClass in h panelGroup
- admin add and delete role page is rewritten and working in #959 so no need to fix it here
- can't reproduce copy trans NPE (not a JSF problem anyway which shouldn't be caused by this PR)

huangp commented:
Cherry picked the fix from #959 now admin add/delete role should work.
